torch/nn/functional.py

```diff
@@ -1635,7 +1649,9 @@ def poisson_nll_loss(input, target, log_input=True, full=False, size_average=Non
 
     """
     if size_average is not None or reduce is not None:
-        reduction = _Reduction.legacy_get_string(size_average, reduce)
+        reduction = _Reduction.legacy_get_string(
+            torch.jit._unwrap_optional(size_average),
```

Comment:
We shouldn't be unwrapping them outside of the call to legacy_get_string. Both size_average and reduce are optional inputs to the function. Also could you add more tests to poisson_nll_loss for different params ?